javaharness/java/arcs/api/ParticleExecutionContextImpl.java

```diff
@@ -4,41 +4,21 @@
 import java.util.HashMap;
 import java.util.Map;
 import java.util.Objects;
-import java.util.Optional;
 import javax.inject.Inject;
 
 public class ParticleExecutionContextImpl implements ParticleExecutionContext {
-  private ParticleLoader particleLoader;
   private PortableJsonParser jsonParser;
   private ArrayList<Particle> particles;
   private HandleFactory handleFactory;
 
   @Inject
   public ParticleExecutionContextImpl(
-      PortableJsonParser jsonParser, ParticleLoader particleLoader, HandleFactory handleFactory) {
+      PortableJsonParser jsonParser, HandleFactory handleFactory) {
```

Comment:
the current demos only use Android particles instantiated in a non-factory manner, but according to the design we will be using the factory instantiation as well. So this code shouldn't be removed.

Reply:
Since we will use the imported code in prod, I really want to keep it as simple as possible. 

Also, there is a design issue here: if we run these compiled particles in ArcsService, on which thread/context will they run? This seems like a big issue to me as it affects Android's system health and robustness. This current implementation doesn't resolve this. So I would rather say we don't support this yet (which is good enough for this release to developers) than have this implementation which might encourage bad implementation.

With non-factory particles, you basically get a free guarantee that a bad particle implementation won't crash your service as it's connected through AIDLs.

Reply:
these particles would run on the same thread/context that the PEC instantiating them.

there is a tricky balance here - the code we've built is an infrastructure among other things, so the current demo is a limited subset of functionality and it doesn't use all the infrastructure's possibilities. this particular aspect will become even more useful as we move towards the post v0 architecture (using the storage layer api). 

there is no way for developers to improperly use this api directly, so it doesn't complicate the api layer and isn't error prone. and it doesn't add dependencies afaict. 

but if you really feel strongly about it, we could remove it and then restore it from history.

Reply:
Let's remove it for now. Without a demo it's hard for me to correctly ship these APIs into the prod system as we need to use a different DI system and the threads management can be dangerous. I can add it back if needed.